### PR TITLE
UTs for HazelcastTimerStore

### PR DESCRIPTION
**Pull request by pzygielo**
# Description
This isn't a bug fix / feature. 

# Testing

### New tests
Check _files changed_ tab.

### Testing Performed
`mvn clean install` completed successfully

### Test suites executed

All included in `mvn clean install` build.

### Testing Environment
Ubuntu 16.04.6 LTS
OpenJDK Runtime Environment (build 1.8.0_222-8u222-b10-1ubuntu1~16.04.1-b10)
Maven 3.6.1

# Documentation

# Notes for Reviewers

Main change in production code: extracted old `listTimers(String[] serverIds)` to new pure function.

### CODE REVIEW

**Comment by Cousjava** [comment]
Jenkins test please

File: appserver/payara-appserver-modules/hazelcast-ejb-timer/src/main/java/fish/payara/ejb/timer/hazelcast/HazelcastTimerStore.java [hunk 1]
```diff
@@ -471,15 +471,13 @@ protected boolean isValidTimerForThisServer(TimerPrimaryKey timerId, RuntimeTime
         return result;
     }
 
-    @Override
-    public String[] listTimers(String[] serverIds) {
+    static String [] listTimers(Collection<HZTimer> timers, String[] serverIds) {
```
Comment (jbee) [inline appserver/payara-appserver-modules/hazelcast-ejb-timer/src/main/java/fish/payara/ejb/timer/hazelcast/HazelcastTimerStore.java]: I feel like the right fix is to add generics to the `pkCache` field `IMap`. That removes the need for cast and method extraction here and in a lot of other places in the file.

File: appserver/payara-appserver-modules/hazelcast-ejb-timer/src/test/java/fish/payara/ejb/timer/hazelcast/HazelcastTimerStoreTestBase.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package fish.payara.ejb.timer.hazelcast;
+
+import java.util.Collection;
+
+public abstract class HazelcastTimerStoreTestBase {
+  public String[] callListTimers(Collection<HZTimer> timers, String[] serverIds) {
```
Comment (jbee) [inline appserver/payara-appserver-modules/hazelcast-ejb-timer/src/test/java/fish/payara/ejb/timer/hazelcast/HazelcastTimerStoreTestBase.java]: `serverIds` can be made a varargs which helps readability in calling methods avoiding the explicit array creation.

File: appserver/payara-appserver-modules/hazelcast-ejb-timer/src/test/java/fish/payara/ejb/timer/hazelcast/HazelcastTimerStoreTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+
+  @Test
+  public void twoTimersForTheSameMemberNameShallBeCountedForTheSameServerId() {
+    String [] serverIds = new String[] { "jb" };
+
+    String [] counts = callListTimers(timers, serverIds);
+
+    assert counts[0].equals("2");
```
Comment (jbee) [inline appserver/payara-appserver-modules/hazelcast-ejb-timer/src/test/java/fish/payara/ejb/timer/hazelcast/HazelcastTimerStoreTest.java]: should use junit's `assertEquals`.

File: appserver/payara-appserver-modules/hazelcast-ejb-timer/src/test/java/fish/payara/ejb/timer/hazelcast/HazelcastTimerStoreTest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  @Test
+  public void noNullsExpectedInCountsForMissingTimers() {
+    String [] serverIds = new String[] { "jb", "ltd", "hz" };
+
+    String [] counts = callListTimers(timers, serverIds);
+
+    for (String count : counts) {
+      assert count != null : "Even for missing timers/server ids no null is expected but rather some representation of zero";
```
Comment (jbee) [inline appserver/payara-appserver-modules/hazelcast-ejb-timer/src/test/java/fish/payara/ejb/timer/hazelcast/HazelcastTimerStoreTest.java]: Should use junit's `assertNotNull`.

**Comment by pzygielo** [comment]
@jbee - thanks Jan for checking. I will rebase and update tests.
I'd be happy to check
> if instead of extracting a method to provide full generic type we add the generics to the field instead as it seems to be expected a fix type everywhere anyway

in separate [at least] commit.

File: appserver/payara-appserver-modules/hazelcast-ejb-timer/src/main/java/fish/payara/ejb/timer/hazelcast/HazelcastTimerStore.java (hunk 1 above)
Comment (pzygielo) [reply appserver/payara-appserver-modules/hazelcast-ejb-timer/src/main/java/fish/payara/ejb/timer/hazelcast/HazelcastTimerStore.java]: I'll prepare separate request with those changes.

**Comment by MarkWareham** [comment]
Jenkins test please

**Pull request merged**